### PR TITLE
Feature/setup refactoring

### PR DESCRIPTION
**Pull request by dev-tim**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,11 +2,16 @@
   "name": "controlcenter",
   "version": "0.0.0",
   "dependencies": {
-    "react": "^0.12.2"
+    "es5-shim": "^4.1.0",
```
Comment (mr-mig) [inline package.json]: Is it used somewhere already?

File: public/webpack.config.js [hunk 2]
```diff
@@ -14,17 +15,21 @@ module.exports = {
...
 	},
 	module: {
 		loaders: [
-            {test: /\.jsx$/, loader: 'jsx-loader?insertPragma=React.DOM&harmony'},
-			{test: /\.scss$/, loader: 'css!autoprefixer!sass?' +
-				'includePaths[]=' + __dirname + '/app/specific' +
-				'&includePaths[]=' + __dirname + '/bower_components/foundation/scss'},
-
+			{test: /\.jsx$/, loader: 'jsx-loader?insertPragma=React.DOM&harmony'},
+			{test: /\.scss$/, loader: ExtractTextPlugin.extract('style','css!autoprefixer!sass?' +
```
Comment (mr-mig) [inline public/webpack.config.js]: Why is ExtractText useful here?

File: public/app/actions/actions.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+'use strict';
```
Comment (mr-mig) [inline public/app/actions/actions.js]: This file can be merged with `index`.
I think there is no sense of using separate index.js if there is only 1 file inside the package

File: public/app/app.jsx [hunk 4]
```diff
@@ -1,39 +1,81 @@
...
-];
-
-function render() {
-    React.render(<Navigation items={navigationItems} />, document.getElementById('navigation'));
-}
-
-render();
+var Reflux = require("reflux");
+
+var appStyle = require('./project-setup/all.scss');
```
Comment (mr-mig) [inline public/app/app.jsx]: Why don't we just use the styles on a per-component basis?
Any comment inside the code with the description of *intent* will be appreciated.

File: public/app/app.jsx [hunk 5]
```diff
@@ -1,39 +1,81 @@
...
+
+var ControlCenter = React.createClass({
+	render: function () {
+		return (
+			<RouteHandler/>
+		)
+	}
+});
+
+var routes = (
```
Comment (mr-mig) [inline public/app/app.jsx]: Can we move Routes and Handlers definitions to a separate file/files? 
As they are supposed to grow...
Or is it not feasible yet?

File: public/app/app.jsx [hunk 6]
```diff
@@ -1,39 +1,81 @@
...
+		<Redirect from="/" to="main"/>
+	</Route>
+);
+
+Router.run(routes, function (Handler, state) {
+	React.render(<Handler params={ state.params } />, document.body);
+});
+
+
+//var addRoom = function() {
```
Comment (mr-mig) [inline public/app/app.jsx]: Why is that stuff commented, but still in the codebase?

File: public/app/common/components/text-input/text-input.jsx [hunk 7]
```diff
@@ -0,0 +1,7 @@
+'use strict';
```
Comment (mr-mig) [inline public/app/common/components/text-input/text-input.jsx]: Is it a demo component? Why is it empty?

File: public/app/common/regions/navigation/navigation.jsx [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+	styles = require("./style.scss"),
+	NavigationItem = require("./nav-item");
+
+var Link = Router.Link;
+
+var Navigation = React.createClass({
+
+	render: function () {
+		return (
+			<div className="cc_navigation">
```
Comment (mr-mig) [inline public/app/common/regions/navigation/navigation.jsx]: :+1:

File: public/app/screens/main/devices/device-list/device-list.jsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+var React = require("react");
+var Device = require('./device');
+var Room = require('./room');
+
+var DeviceList = React.createClass({
```
Comment (mr-mig) [inline public/app/screens/main/devices/device-list/device-list.jsx]: I would like to reconsider the indentation rules: 4 spaces for a tab are too much.

File: public/app/screens/main/devices/device-list/device/device.jsx [hunk 10]
```diff
@@ -7,16 +6,16 @@ var Device = React.createClass({
         return (
             <div className="row">
                 <div className="small-4 column">
-          {this.props.device.title}
+          adsfas
```
Comment (mr-mig) [inline public/app/screens/main/devices/device-list/device/device.jsx]: What's that?

File: public/app/screens/main/devices/device-list/room/room.jsx [hunk 11]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+var React = require('react');
+//How to we communicate subling components
```
Comment (mr-mig) [inline public/app/screens/main/devices/device-list/room/room.jsx]: Is it a question? Is it a TODO? Is it an example?

File: public/app/screens/main/devices/device-list/room/room.jsx [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+var Device = require('../device');
+
+var Room = React.createClass({
+	render: function () {
+		return (
+			<div>
+				<h3>Hello</h3>
+        <Device />
```
Comment (mr-mig) [inline public/app/screens/main/devices/device-list/room/room.jsx]: What's with the indentation?

File: .merge_file_TJDZmK [hunk 13]
```diff
@@ -0,0 +1,3 @@
+.ololo{
```
Comment (larsblumberg) [inline .merge_file_TJDZmK]: Guess you added this file accidentally

File: .merge_file_Z57D22 [hunk 14]
```diff
@@ -0,0 +1,3 @@
+.ololo{
```
Comment (larsblumberg) [inline .merge_file_Z57D22]: Guess you added this file accidentally

File: .merge_file_hKTdyt [hunk 15]
```diff
@@ -0,0 +1,3 @@
+.ololo{
```
Comment (larsblumberg) [inline .merge_file_hKTdyt]: Guess you added this file accidentally

File: public/app/actions/actions.js [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+var Reflux = require('reflux');
+
+var actions = Reflux.createActions([
```
Comment (larsblumberg) [inline public/app/actions/actions.js]: We will have tons of actions. It might be good to have multiple, separated action files. See https://github.com/yetu/controlcenter/pull/2/files

However this can be refactored later.

File: public/app/app.jsx [hunk 17]
```diff
@@ -1,39 +1,81 @@
...
+var Router = require('react-router');
+var RouteHandler = Router.RouteHandler;
+var Route = Router.Route;
+var DefaultRoute = Router.DefaultRoute;
+var NotFoundRoute = Router.NotFoundRoute;
+var Redirect = Router.Redirect;
+var Link = Router.Link;
+
+var MainScreen = require('./screens/main').MainScreen;
+var DevicesRegion = require('./screens/main').DevicesRegion;
```
Comment (larsblumberg) [inline public/app/app.jsx]: I personally don't like the `region` name. Would I'd rather like to call it `section` as facebook does it as well.

File: public/app/app.jsx (hunk 6 above)
Comment (dev-tim) [inline public/app/app.jsx]: Because we want to use this model for devices implementation

File: public/app/app.jsx (hunk 17 above)
Comment (mr-mig) [inline public/app/app.jsx]: :+1: for section name. 
Moreover, it is semantically aligned with HTML5 `<section>`

File: public/app/app.jsx (hunk 5 above)
Comment (dev-tim) [inline public/app/app.jsx]: Yes we can. Later.

File: public/app/app.jsx.BACKUP.32588.jsx [hunk 18]
```diff
@@ -0,0 +1,96 @@
+'use strict';
+
+var React = require('react');
+var Reflux = require("reflux");
+
+<<<<<<< HEAD
```
Comment (larsblumberg) [inline public/app/app.jsx.BACKUP.32588.jsx]: Artem, I want you to review your own pull request first before presenting it to others.

File: public/app/components/common/text-input/text-input.jsx [hunk 19]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+var React = require("react");
+
+var TextInput = React.createClass({});
```
Comment (larsblumberg) [inline public/app/components/common/text-input/text-input.jsx]: empty `render` method?

File: public/app/app.jsx (hunk 4 above)
Comment (dev-tim) [inline public/app/app.jsx]: We do use it. There is an import of app component where I add global styles to the root component.

File: public/app/actions/actions.js (hunk 3 above)
Comment (dev-tim) [inline public/app/actions/actions.js]: There is a possibility that there can be more modules, likes actions pipeline etc.

File: public/app/screens/main/devices/device-list/device-list.jsx (hunk 9 above)
Comment (larsblumberg) [inline public/app/screens/main/devices/device-list/device-list.jsx]: Why does a device list encapsulate a room? Why do we need this layer of abstraction?

File: public/webpack.config.js (hunk 2 above)
Comment (dev-tim) [inline public/webpack.config.js]: It's experiment, we can use just a simple loader and attache styles to the DOM (what it not so good for prod case)

File: public/app/screens/main/devices/device-list/device-list.jsx [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+	render: function () {
+		return (
+				<Room  />
```
Comment (larsblumberg) [inline public/app/screens/main/devices/device-list/device-list.jsx]: Why do you use real tabs instead of spaces? Is this our convention?

File: public/app/screens/main/devices/device-list/room/room.jsx (hunk 11 above)
Comment (larsblumberg) [inline public/app/screens/main/devices/device-list/room/room.jsx]: Please check grammar of your comment, add TODO

Comment (dev-tim) [inline public/app/screens/main/devices/device-list/room/room.jsx]: Sure it's a question, because it's not implemented here.

File: public/app/screens/main/devices/devices.jsx~HEAD [hunk 21]
```diff
@@ -0,0 +1,19 @@
+'use strict';
```
Comment (larsblumberg) [inline public/app/screens/main/devices/devices.jsx~HEAD]: Why did you add this file?

File: public/app/stores/device-store/index.js [hunk 22]
```diff
@@ -0,0 +1,2 @@
+'use strict';
```
Comment (larsblumberg) [inline public/app/stores/device-store/index.js]: No need to call a folder `device-store` inside a `store` folder

File: public/app/app.jsx (hunk 17 above)
Comment (dev-tim) [inline public/app/app.jsx]: Let's discuss it, we agreed on Friday on regions, but I don't mind on sections as well

**Comment by larsblumberg** [comment]
Artem, I would like you to review your own pull request first before presenting it to others.

You've left a lot of merge artefacts in the commit. I'm unhappy to see such a high amount of them.

File: public/app/actions/actions.js (hunk 16 above)
Comment (dev-tim) [inline public/app/actions/actions.js]: I would also vote for separate PR.

File: public/app/app.jsx (hunk 17 above)
Comment (dev-tim) [inline public/app/app.jsx]: I suggest to make it a separate PR.

File: public/app/app.jsx (hunk 6 above)
Comment (mr-mig) [inline public/app/app.jsx]: And why is it commented?

File: public/app/screens/main/devices/device-list/device-list.jsx (hunk 9 above)
Comment (dev-tim) [inline public/app/screens/main/devices/device-list/device-list.jsx]: Actually it's just a placeholder since we don't have logic there, here I wanted to test interaction between components. Probably you are right, we might not need it in future.

File: package.json (hunk 1 above)
Comment (dev-tim) [inline package.json]: Not yet, but required by reflux

File: public/app/common/components/text-input/text-input.jsx (hunk 7 above)
Comment (dev-tim) [inline public/app/common/components/text-input/text-input.jsx]: Yep it's placeholder for general componets

File: public/app/common/regions/navigation/navigation.jsx (hunk 8 above)
Comment (dev-tim) [inline public/app/common/regions/navigation/navigation.jsx]: ?

File: public/app/screens/main/devices/device-list/room/room.jsx (hunk 12 above)
Comment (dev-tim) [inline public/app/screens/main/devices/device-list/room/room.jsx]: Something wrong ((

**Comment by dev-tim** [comment]
Lars, you see these artefacts because quite huge rebase, however idea of pull requests aims to improve code quality by making comments
